Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -2031,29 +2034,34 @@ extension SourceKitServer {
         position: req.position
       )
     )
-    guard let symbol = symbols.first, let index = await self.workspaceForDocument(uri: req.textDocument.uri)?.index
-    else {
+    guard let index = await self.workspaceForDocument(uri: req.textDocument.uri)?.index else {
       return nil
     }
     // For call hierarchy preparation we only locate the definition
-    guard let usr = symbol.usr else { return nil }
+    let usrs = symbols.compactMap(\.usr)
 
     // Only return a single call hierarchy item. Returning multiple doesn't make sense because they will all have the
     // same USR (because we query them by USR) and will thus expand to the exact same call hierarchy.
-    // Also, VS Code doesn't seem to like multiple call hierarchy items being returned and fails to display any results
-    // if they are, failing with `Cannot read properties of undefined (reading 'map')`.
-    guard let definition = index.definitionOrDeclarationOccurrences(ofUSR: usr).first else {
-      return nil
-    }
-    guard let location = indexToLSPLocation(definition.location) else {
-      return nil
+    var callHierarchyItems = usrs.compactMap { (usr) -> CallHierarchyItem? in
+      guard let definition = index.primaryDefinitionOrDeclarationOccurrence(ofUSR: usr) else {
+        return nil
+      }
+      guard let location = indexToLSPLocation(definition.location) else {
+        return nil
+      }
+      return self.indexToLSPCallHierarchyItem(
+        symbol: definition.symbol,
+        moduleName: definition.location.moduleName,
+        location: location
+      )
     }
-    let callHierarchyItem = self.indexToLSPCallHierarchyItem(
-      symbol: definition.symbol,
-      moduleName: definition.location.moduleName,
-      location: location
-    )
-    return [callHierarchyItem]
+
+    // Ideally, we should show multiple symbols. But VS Code fails to display call hierarchies with multiple root items,
+    // failing with `Cannot read properties of undefined (reading 'map')`. Pick the first one.
+    callHierarchyItems = Array(callHierarchyItems.prefix(1))
```

Comment:
`[callHierarchyItems.first]` would be shorter, any reason to prefer prefix?

Reply:
It would be `[callHierarchyItem.first].compactMap { $0 }`. Otherwise you get an array with a single optional not an array with 0 or 1 items containing non-optionals. And then I thought `.prefix` is nicer.